src/Aquifer.API/Endpoints/Notifications/List/Response.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Aquifer.API.Common.Dtos;
+using Aquifer.Data.Entities;
+
+namespace Aquifer.API.Endpoints.Notifications.List;
+
+public sealed class Response
+{
+    public required NotificationKind Kind { get; init; }
```

Comment:
This uses a discriminated union pattern (though somewhat forced because C# doesn't officially support discriminated unions).  The `Comment` or `HelpDocument` sub-DTO will only be populated if the `Kind` here is `Comment` or `HelpDocument`, respectively.